mailbox/api/src/test/java/org/apache/james/mailbox/PathDelimiterTest.java

```diff
@@ -0,0 +1,438 @@
+            .isEqualTo("");
+    }
+
+    @Test
+    public void removeHeadingSeparatorShouldDoNothingWhenNoSeparator() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER
+            .removeHeadingSeparator("aa"))
+            .isEqualTo("aa");
+    }
+
+    @Test
+    public void removeHeadingSeparatorShouldDoNothingWhenMiddleSeparator() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER
+            .removeHeadingSeparator("a.a"))
+            .isEqualTo("a.a");
+    }
+
+    @Test
+    public void removeHeadingSeparatorShouldReturnBaseNameWhenStartingByDelimiter() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER
+            .removeHeadingSeparator(".aa"))
+            .isEqualTo("aa");
+    }
+
+    @Test
+    public void removeHeadingSeparatorShouldRemoveSeparatorOnlyOnce() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER
+            .removeHeadingSeparator("..aa"))
+            .isEqualTo(".aa");
+    }
+
+    @Test
+    public void removeHeadingSeparatorShouldDoNothingWhenSeparatorAtTheEnd() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER
+            .removeHeadingSeparator("aa."))
+            .isEqualTo("aa.");
+    }
+
+    @Test
+    public void getHierarchyLevelsShouldBeOrdered() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.getHierarchyLevels("inbox.folder.subfolder")
+            .collect(Guavate.toImmutableList()))
+            .containsExactly(
+                "inbox",
+                "inbox.folder",
+                "inbox.folder.subfolder");
+    }
+
+    @Test
+    public void getHierarchyLevelsShouldReturnNameWhenOneLevel() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.getHierarchyLevels("inbox")
+            .collect(Guavate.toImmutableList()))
+            .containsExactly(
+                "inbox");
+    }
+
+    @Test
+    public void getHierarchyLevelsShouldReturnNameWhenEmptyName() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.getHierarchyLevels("")
+            .collect(Guavate.toImmutableList()))
+            .containsExactly("");
+    }
+
+    @Test
+    public void getHierarchyLevelsShouldReturnNameWhenNullName() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.getHierarchyLevels(null)
+            .map(Optional::ofNullable)
+            .collect(Guavate.toImmutableList()))
+            .containsExactly(Optional.empty());
+    }
+
+    @Test
+    public void toPatternShouldQuoteDelimiter() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.toPattern())
+            .isEqualTo("\\Q.\\E");
+    }
+
+    @Test
+    public void asStringShouldReturnStringValueOfDelimiter() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.asString())
+            .isEqualTo(".");
+    }
+
+    @Test
+    public void isUndefinedShouldReturnFalseWhenNotZero() {
+        assertThat(MailboxConstants.DEFAULT_DELIMITER.isUndefined())
+            .isFalse();
+    }
+
+    @Test
+    public void isUndefinedShouldReturnTrueWhenZero() {
```

Comment:
that's not enough, when it's undefined, it should not try to split a string into a parts

Reply:
Will handle that with API + 2 PathDelimiter implementation